src/data/set/intervals/group.lean

```diff
@@ -98,4 +101,107 @@ end
 
 end linear_ordered_add_comm_group
 
+/-! ### Lemmas about disjointness of translates of intervals -/
+section pairwise_disjoint
+
+section ordered_comm_group
+
+variables [ordered_comm_group α] (a b : α)
+
+@[to_additive]
+lemma pairwise_disjoint_Ioc_mul_zpow  :
+  pairwise (disjoint on (λ n : ℤ, Ioc (a * b ^ n) (a * b ^ (n + 1)))) :=
+begin
+  simp_rw [function.on_fun, set.disjoint_iff],
+  intros m n hmn x hx,
+  apply hmn,
+  by_cases hb : 1 < b,
+  { have i1 := hx.1.1.trans_le hx.2.2,
+    have i2 := hx.2.1.trans_le hx.1.2,
+    rw [mul_lt_mul_iff_left, zpow_lt_zpow_iff hb, int.lt_add_one_iff] at i1 i2,
+    exact le_antisymm i1 i2 },
+  { -- case 0 ≮ b : vacuous but true
+    have : ∀ n : ℤ, Ioc (a * b ^ n) (a * b ^ (n + 1)) = ∅,
+    { refine λ n, Ioc_eq_empty_iff.mpr _,
+      rwa [mul_lt_mul_iff_left, ←mul_one (b ^ n), zpow_add_one, mul_lt_mul_iff_left] },
+    simp_rw [this, empty_inter] at hx,
+    exact hx.elim }
+end
+
+@[to_additive]
+lemma pairwise_disjoint_Ico_mul_zpow :
+  pairwise (disjoint on (λ n : ℤ, Ico (a * b ^ n) (a * b ^ (n + 1)))) :=
+begin
+  simp_rw [function.on_fun, set.disjoint_iff],
+  intros m n hmn x hx,
+  apply hmn,
+  by_cases hb : 1 < b,
```

Comment:
Can you try to directly derive `1 < b` from `hx`? something like `rw [add_lt_add_iff_right, some_pow_lemma] at hx`. I have a hunch that it will be much shorter.

Reply:
The `[some_pow_lemma]` ends up being exactly what I wrote in the second case of the `rcases` before. The best I could do is 
```
begin
  simp_rw [function.on_fun, set.disjoint_iff],
  intros m n hmn x hx,
  apply hmn,
  have hb : 1 < b,
  { have ne : Ioc (a * b ^ m) (a * b ^ (m + 1)) ≠ ∅, 
      from nonempty_iff_ne_empty.mp (nonempty_of_mem $ mem_of_mem_inter_left hx),
    contrapose! ne,
    rwa [Ioc_eq_empty_iff, mul_lt_mul_iff_left, ←mul_one (b ^ m), zpow_add_one, 
      mul_lt_mul_iff_left] },
  have i1 := hx.1.1.trans_le hx.2.2,
  have i2 := hx.2.1.trans_le hx.1.2,
  rw [mul_lt_mul_iff_left, zpow_lt_zpow_iff hb, int.lt_add_one_iff] at i1 i2,
  exact le_antisymm i1 i2
end
```
where the only net saving is omitting the one-line comment. (The problem is that for some strange reason `Ioc_eq_empty` won't rewrite on ` ≠ ∅`, but  `ne.def` won't convert it into `not (_ = ∅)`.)

Reply:
No, wait, I have a better approach now.

Reply:
> Can you try to directly derive 1 < b from hx

OK, I've now done it that way. (It could be rendered shorter still by not explicitly giving the type of `hb` but that would render the proof less readable so I don't think it's a gain.)